src/api/app/services/architectures_controller_service/architecture_updater.rb

```diff
@@ -1,16 +1,13 @@
 module ArchitecturesControllerService
   class ArchitectureUpdater
-    def initialize(archs)
-      @archs = archs
-      @all_valid = true
+    def initialize(params)
+      @archs = params.require(:archs).permit!
+      @all_valid = false
     end
 
     def call
-      @archs.each do |name, value|
-        arch = Architecture.find_by(name: name)
-        arch.available = value
-        @all_valid &&= arch.save
-      end
+      @all_valid = @archs.to_h.map { |name, value| Architecture.where(name: name).update(available: value) }.flatten.all?(&:valid?)
```

Comment:
this was more efficient before... :sob: 

I would try to do less queries in the database instead... :thinking:

Reply:
This table has as maximum 20 to 30 architectures. I doubt this has any relevant performance impact.

Reply:
hmm, IMO it *are* less database queries.

I didn't check in the log, but before you had
SELECT * from architectures limit 1;
UPDATE architectures where id=X;

and now you have 
UPDATE architectures where name=X;

Reply:
right @coolo, @Ana06 could you please check it again and tell us concretely what are your concerns?

thank you

Reply:
@coolo I think what @Ana06 means is that the `.all?(&:valid?)` causes another iteration, while before we would evaluate the validation result right away.

As a sidenote `all?` stops the iteration on the first falsy element.

Reply:
> @coolo I think what @Ana06 means is that the `.all?(&:valid?)` causes another iteration, while before we would evaluate the validation result right away.

but does it generates a new SQL query? looking in the log it doesnt look like. 

I think the step towards idiomatic code is the right decision here. Even if in a microbenchmarking the older code was faster, The number of architectures won't grow up in a way that we have to sacrifice code readability for speed.

Reply:
I don't consider your code more readable to be honest.

For once: you call `valid?` on what exactly? On an architecture? On a string? On a hash element?

Reply:
> I don't consider your code more readable to be honest.

We can consider that more idiomatic code is more readable, since it's the kind of code that we are used to ready daily.

Mutation + ```&&=``` is for sure less idomatic ruby than method chaining.

> For once: you call `valid?` on what exactly? On an architecture? On a string? On a hash element?

The good thing about avoiding mutation and method chaining is that you can easily paste the one line in your console and remove the last call to see it

Reply:
... if it didn't had side effects.

Reply:
> ... if it didn't had side effects.

side effects and mutations aren't the same thing, but you know it.

Reply:
you are forgetting again Ruby... you go through the array many more times. I would leave it as it was before. I don't see the improvement.

What I meant with the queries is that it may be possible to update all the architectures at the same time.

Reply:
Hi @Ana06 
the other implementation that we came up with was:

```
...

def call
 @all_valid = @archs.to_h.map do |name, value|
                        arch = Architecture.find_by(name: name)	
                        arch.available = value	
                        arch.save
                      end
  self
 end

def valid?
 @all_valid.all?
end
```

Reply:
@vpereira I don't think you need to return `self`, but appart from that this suggestion LGTM :smile:

Reply:
leave my self alone

Reply:
@vpereira @Ana06 I've updated the PR again. Sorry for the long delay.

 > Ana06 12 days ago Member
>
> What I meant with the queries is that it may be possible to update all the architectures at the same time.

We could do that with `update_all` but then the validations wouldn't run.